torch/distributed/fsdp/flat_param.py

```diff
@@ -794,10 +794,11 @@ def init_flat_param_attributes(self) -> None:
         if self._offload_params:
             p_assert(
                 flat_param.device == cpu_device,
```

Comment:
We simplify this assert message because it is an internal assert. Accidentally moving the model to GPU after FSDP constructor will be caught before in `_check_flat_params_on_expected_device()`.